packages/core/src/routes-me/social.ts

```diff
@@ -0,0 +1,119 @@
+ * Thus it does NOT support connectors rely on the session (jti based) storage. E.g. Apple connector and all standard connectors.
+ * This is because Logto Cloud AC only supports Google and GitHub social sign-in, both of which do not rely on session storage.
+ */
+export default function socialRoutes<T extends AuthedMeRouter>(
+  ...[router, tenant]: RouterInitArgs<T>
+) {
+  const {
+    libraries: {
+      connectors: { getLogtoConnectorById },
+    },
+    queries: {
+      users: { findUserById, updateUserById },
+    },
+  } = tenant;
+
+  router.post(
+    '/social-authorization-uri',
+    koaGuard({
+      body: object({ connectorId: string(), state: string(), redirectUri: string() }),
+    }),
+    async (ctx, next) => {
+      const { connectorId, state, redirectUri } = ctx.guard.body;
+      assertThat(state && redirectUri, 'session.insufficient_info');
+
+      const connector = await getLogtoConnectorById(connectorId);
+      assertThat(connector.type === ConnectorType.Social, 'connector.unexpected_type');
+
+      const {
+        headers: { 'user-agent': userAgent },
+      } = ctx.request;
+
+      const redirectTo = await connector.getAuthorizationUri(
+        {
+          state,
+          redirectUri,
+          connectorId,
+          connectorFactoryId: connector.metadata.id,
+          /**
+           * Passing empty jti only works for connectors not relying on session storage.
+           * E.g. Google and GitHub
+           */
+          jti: '',
+          headers: { userAgent },
+        },
+        /**
+         * Same as above, passing `notImplemented` only works for connectors not relying on session storage.
+         * E.g. Google and GitHub
+         */
+        notImplemented
+      );
+
+      ctx.body = { redirectTo };
+      ctx.status = 200;
+
+      return next();
+    }
+  );
+
+  router.post(
+    '/link-social-identity',
+    koaGuard({
+      body: object({
+        connectorId: string(),
+        connectorData: record(string(), unknown()),
+      }),
+    }),
+    async (ctx, next) => {
+      const { id: userId } = ctx.auth;
+      const { connectorId, connectorData } = ctx.guard.body;
+      const connector = await getLogtoConnectorById(connectorId);
+
+      assertThat(
+        connector.type === ConnectorType.Social,
+        new RequestError({
+          code: 'session.invalid_connector_id',
+          status: 422,
+          connectorId,
+        })
+      );
+
+      /**
+       * Same as above, passing `notImplemented` only works for connectors not relying on session storage.
+       * E.g. Google and GitHub
+       */
+      const userinfo = await connector.getUserInfo(connectorData, notImplemented, {
+        get: notImplemented,
+        set: notImplemented,
+      });
+      const user = await findUserById(userId);
+
+      await updateUserById(userId, {
+        identities: {
+          ...user.identities,
+          [connectorId]: {
```

Comment:
should be `connector.metadata.target`?

Reply:
Should we also ensure that the key `connector.metadata.target` does not exist in `user.identities`? (although this would rarely happen)

Reply:
You are right, this should be `target`. And I will add a check before binding the identity.
Thanks!